### ISSUE TITLE
Add details on generating a local copy of the MDF documentation 

### ISSUE THREAD

**Issue by pgleeson** [ModECI/MDF#357]
Can be added at the end of https://github.com/ModECI/MDF/blob/main/docs/sphinx/source/api/Installation.md, which is generated and deployed at https://mdf.readthedocs.io/en/latest/api/Installation.html

Will be something like:
```
./test_all.sh # tests the package, runs docs/generate.py, to copy the main README and example readmes into docs/sphinx/source/api
cd docs/sphinx
make html # creates build/html/index.html etc.
```

**Comment by Monsurat-Onabajo** [comment]
Hello, i was checking the issue and i unassigned myself mistakenly, can you please assign me back, thank you

**Comment by pgleeson** [comment]
No problem @Monsurat-Onabajo, reassigned you.

**Comment by sanjayankur31** [comment]
Getting make to work on windows/powershell:

https://superuser.com/questions/808807/using-make-from-windows-powershell

**Comment by pgleeson** [comment]
I've added your branches to the readthedocs project and now they will generate automatically, e.g.

@Monsurat-Onabajo https://mdf.readthedocs.io/en/add_contributors/api/Contributors.html
@kusanele https://mdf.readthedocs.io/en/update_readme_links/api/export_format/MDF/MDF.html

Note they take ~5 mins to generate after a commit. 
 
Nevertheless, it would be much better to get a version of the sphinx documentation building locally for testing.

**Comment by Monsurat-Onabajo** [comment]
Noted, thank you

**Comment by Monsurat-Onabajo** [comment]
Hello, i have installed make and sphinx on my computer locally but anytime i want to use sphinx to generate the documentation offline using html, i kept on getting this error (i have tried the same with another file and it is working well). And also when i try to use sphinx-activate, i kept on getting that there is conf.py file already in the folder, and i cant seem to find a way to work around it.
![IMG_20221213_131433](https://user-images.githubusercontent.com/102332122/207316952-2c2475cd-71fd-4705-a0e7-11b4ad8a787c.jpg)

**Comment by pgleeson** [comment]
I believe root of this problem is that the building of the docs is not working on python 3.10. 

This is possibly caused by this line in the requirements which sets the version of sphinx to use:
https://github.com/ModECI/MDF/blob/b9b60f4f00a286c530b4baac2c1567e7ad483bb0/setup.py#L17

To check this I've created a new branch https://github.com/ModECI/MDF/tree/feature/test_doc_generation and added the following lines to the GitHub Actions test script:
https://github.com/ModECI/MDF/blob/c5d2403d3dd49d38a59259808d1ef7d56d34b9f2/.github/workflows/ci.yml#L68-L73

This is tested in the multiple combinations of platforms and python versions we test on, see: https://github.com/ModECI/MDF/actions/runs/3693811822

Clearly py 3.10 is failing on all platforms. 

Rather than trying to fix the documentation for 3.10 just yet, it would be better for you both to get a virtual environment based on python 3.9 installed using conda, then install mdf and all dependencies in that using `python -m pip install .[all]` and test generation of the docs. 

When you have that working (might require reading up on using conda etc on Windows) and made some locally tested commits to the documentation, we can see how much work it would be to update to a later version of sphinx which is compatible with py 3.10

**Comment by Monsurat-Onabajo** [comment]
Ok, thank you... I tried commenting out the 'InstanceAttributeDocumenter' library and i was able to generate the documentation offline and it is working perfectly fine, but i dont think that is good for long term, i will definitely try running it on python 3.9 and see how it will go, thank you
![IMG_20221213_203316](https://user-images.githubusercontent.com/102332122/207585251-e526ba9e-7a09-450b-ab20-8dc8ff744a3e.jpg)

**Comment by Monsurat-Onabajo** [comment]
> I believe root of this problem is that the building of the docs is not working on python 3.10.
> 
> This is possibly caused by this line in the requirements which sets the version of sphinx to use:
> 
> https://github.com/ModECI/MDF/blob/b9b60f4f00a286c530b4baac2c1567e7ad483bb0/setup.py#L17
> 
> To check this I've created a new branch https://github.com/ModECI/MDF/tree/feature/test_doc_generation and added the following lines to the GitHub Actions test script:
> 
> https://github.com/ModECI/MDF/blob/c5d2403d3dd49d38a59259808d1ef7d56d34b9f2/.github/workflows/ci.yml#L68-L73
> 
> This is tested in the multiple combinations of platforms and python versions we test on, see: https://github.com/ModECI/MDF/actions/runs/3693811822
> 
> Clearly py 3.10 is failing on all platforms.
> 
> Rather than trying to fix the documentation for 3.10 just yet, it would be better for you both to get a virtual environment based on python 3.9 installed using conda, then install mdf and all dependencies in that using `python -m pip install .[all]` and test generation of the docs.
> 
> When you have that working (might require reading up on using conda etc on Windows) and made some locally tested commits to the documentation, we can see how much work it would be to update to a later version of sphinx which is compatible with py 3.10
them
Hello, i have tried working on it exactly as you suggested but i am still unable to generate the document offline as a result of that issue, but i discovered that when comment the library out, i was able to generate the documentation.i am reading more about sphinx documentation in order to know if it is system error or not
